### DISCUSSION TITLE
Acces MLEADER content

### DISCUSSION THREAD

**Discussion by cristobalinho** [mozman/ezdxf#916] [Q&A] [answered]
Hello everybody!
I need to access MLEADER entity, and after I read the documentation, I found it was a very complex entity but anyway I could be able to access the text content by:
`def FindTextMleader(Mleader):
    for tag in Mleader.tags:
        group_code, value = tag
        if group_code == 304: 
            return value`
 I need to know if there is a way to obtain the target point of the MLEADER, or the insertion point. Thanks in advance!

**Comment by mozman** [comment]
The `MultiLeader` entity has no attribute `tags`. 

Printing the content and the line vertices of MULTILEADER entities that have a MTEXT context:

```Python
import ezdxf

doc = ezdxf.readfile("your.dxf")
msp = doc.modelspace()
for mleader in msp.query("MULTILEADER"):
    # skip BLOCK context
    if mleader.has_block_content:
        continue

    print(mleader.context.mtext.default_content)
    # maybe multiple leaders (left and/or right)
    for leader in mleader.context.leaders:
        # maybe multiple lines per leader
        for line in leader.lines:
            # the first vertex is the target point
            print(line.vertices[0])

```
The docs and further links for MULTILEADER/MLEADER can be found here: https://ezdxf.mozman.at/docs/dxfentities/mleader.html

**Reply by cristobalinho** [reply]
Thank you for your quick response. I don't know why, but the mleader entities I'm working with, has tags attributes. The class for this entities is 'ezdxf.entities.mleader.MultiLeader'.
However using the code above raise the error: " 'MultiLeader' object has no attribute 'has_block_content' ". I don't know why this error could happened. Thank you again!

**Reply by cristobalinho** [reply]
I was already able to find the vertex in mleader.tags. But I still don't understand why this mleader entity is different.

**Reply by mozman** [reply, marked as answer]
You are using an old version of `ezdxf`, the current version is v1.0.3, check your version by:

```Python
import ezdxf; print(ezdxf.__version__)
```

**Reply by cristobalinho** [reply]
Thanks!!
